File: LabourBudgetCalculator/Models/CalculationResult.cs
using System;
using System.Collections.Generic;

namespace LabourBudgetCalculator.Models
{
    [Serializable]
    public class CalculationResult
    {
        public decimal TotalLabourCost { get; set; }
        public decimal TotalTravelCost { get; set; }
        public decimal TotalExpenses { get; set; }
        public decimal GrandTotal { get; set; }
        public int TotalDays { get; set; }
        public int TotalLabourHours { get; set; }
        public int TotalTravelHours { get; set; }

        // Daily breakdown
        public List<DayDetail> DayDetails { get; set; } = new List<DayDetail>();
    }

    [Serializable]
    public class DayDetail
    {
        public int DayNumber { get; set; }
        public DayOfWeek DayOfWeek { get; set; }
        public CalculationDayType Type { get; set; }
        public int RegularLabourHours { get; set; }
        public int OvertimeLabourHours { get; set; }
        public int PremiumLabourHours { get; set; }
        public int TotalLabourHours { get; set; }
        public int RegularTravelHours { get; set; }
        public int OvertimeTravelHours { get; set; }
        public int PremiumTravelHours { get; set; }
        public int TotalTravelHours { get; set; }
        public decimal LabourCost { get; set; }
        public decimal TravelCost { get; set; }
        public decimal HotelCost { get; set; }
        public decimal PerDiem { get; set; }
        public decimal MileageCost { get; set; }
        public decimal RentalCarCost { get; set; }
        public decimal AirfareCost { get; set; }
        public decimal TotalDayCost { get; set; }
    }

    public enum CalculationDayType
    {
        WorkDay,
        TravelTo,
        TravelFrom,
        None
    }
}

File: src/models/CalculationResult.ts
export enum CalculationDayType {
  WorkDay = 'WorkDay',
  TravelTo = 'TravelTo',
  TravelFrom = 'TravelFrom',
  None = 'None'
}

export interface DayDetail {
  dayNumber: number;
  dayOfWeek: number; // 0 = Sunday, 1 = Monday, etc.
  type: CalculationDayType;
  regularLabourHours: number;
  overtimeLabourHours: number;
  premiumLabourHours: number;
  totalLabourHours: number;
  regularTravelHours: number;
  overtimeTravelHours: number;
  premiumTravelHours: number;
  totalTravelHours: number;
  labourCost: number;
  travelCost: number;
  hotelCost: number;
  perDiem: number;
  mileageCost: number;
  rentalCarCost: number;
  airfareCost: number;
  totalDayCost: number;
}

export interface CalculationResult {
  totalLabourCost: number;
  totalTravelCost: number;
  totalExpenses: number;
  grandTotal: number;
  totalDays: number;
  totalLabourHours: number;
  totalTravelHours: number;
  
  // Daily breakdown
  dayDetails: DayDetail[];
} 